src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -464,11 +464,9 @@ private void processVariableDef(DetailAST ast, DetailAST nextToken) {
      * @return true if policy is violated and warning should be raised; false otherwise
      */
     private boolean isViolatingEmptyLineBetweenFieldsPolicy(DetailAST detailAST) {
-        return allowNoEmptyLineBetweenFields
-                    && detailAST.getType() != TokenTypes.VARIABLE_DEF
-                    && detailAST.getType() != TokenTypes.RCURLY
-                || !allowNoEmptyLineBetweenFields
-                    && detailAST.getType() != TokenTypes.RCURLY;
+        return detailAST.getType() != TokenTypes.RCURLY
```

Comment:
Reworked conditionals to kill mutation. Both sections end with `rcurly` so I put that at front and simplified the `allowNoEmptyLineBetweenFields || !allowNoEmptyLineBetweenFields`.